File: app/src/main/java/dev/ragnarok/fenrir/adapter/SideDrawerCategotiesAdapter.java
package dev.ragnarok.fenrir.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import dev.ragnarok.fenrir.R;
import dev.ragnarok.fenrir.model.SideDrawerCategory;

public class SideDrawerCategotiesAdapter extends RecyclerView.Adapter<SideDrawerCategotiesAdapter.ViewHolder> {

    private List<SideDrawerCategory> data;

    public SideDrawerCategotiesAdapter(List<SideDrawerCategory> data) {
        this.data = data;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_drawer_category, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        SideDrawerCategory category = data.get(position);

        holder.checkBox.setText(category.getTitle());
        holder.checkBox.setOnCheckedChangeListener(null);
        holder.checkBox.setChecked(category.isChecked());

        holder.checkBox.setOnCheckedChangeListener((buttonView, isChecked) -> category.setChecked(isChecked));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public void setData(List<SideDrawerCategory> data) {
        this.data = data;
        notifyDataSetChanged();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        final CheckBox checkBox;

        public ViewHolder(View itemView) {
            super(itemView);
            checkBox = itemView.findViewById(R.id.item_drawer_category_check);
        }
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/fragment/navigation/sidedraweredit/SideDrawerCategotiesAdapter.kt
package dev.ragnarok.fenrir.fragment.navigation.sidedraweredit

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.checkbox.MaterialCheckBox
import dev.ragnarok.fenrir.R
import dev.ragnarok.fenrir.model.SideDrawerCategory

class SideDrawerCategotiesAdapter(private var data: List<SideDrawerCategory>) :
    RecyclerView.Adapter<SideDrawerCategotiesAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.item_drawer_category, parent, false)
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val category = data[position]
        holder.checkBox.setText(category.getTitle())
        holder.checkBox.setOnCheckedChangeListener(null)
        holder.checkBox.isChecked = category.isChecked()
        holder.checkBox.setOnCheckedChangeListener { _: CompoundButton?, isChecked: Boolean ->
            category.setChecked(isChecked)
        }
    }

    override fun getItemCount(): Int {
        return data.size
    }

    fun setData(data: List<SideDrawerCategory>) {
        this.data = data
        notifyDataSetChanged()
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val checkBox: MaterialCheckBox = itemView.findViewById(R.id.item_drawer_category_check)
    }
}